### PR TITLE
Create virtual env automatically

### LINKED ISSUE #23: Ensure venv is automatically created when the plugin is enabled.
This refers to the venv the language server is installed in, which must be manually specified and created at the moment.

### PR DESCRIPTION
**Pull request by andraskvr**
Closes #23

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -150,10 +145,35 @@ export function activate(context: ExtensionContext) {
...
+		}
+		const venvDir = Uri.joinPath(context.globalStorageUri, ".env").fsPath;
+		const venvPath = Uri.joinPath(context.globalStorageUri, ".env", "bin", "python3").fsPath;
+
+		if (!fs.existsSync(venvDir)) {
+			const venvProcess = cp.spawnSync("python3", ["-m", "venv", venvDir ]);
+			if (venvProcess.status != 0) {
+				window.showErrorMessage(`Virtual env creation at ${venvDir} failed with code ${venvProcess.status}, ${venvProcess.stderr}`);
+			}
+			const child = cp.spawnSync(venvPath, ["-m", "pip", "install", "--pre", "inmantals"]);
```
Comment (sanderr) [inline src/extension.ts:164]: Why `--pre`?

**Review by sanderr — CHANGES_REQUESTED** [review]

File: src/extension.ts (hunk 1 above)
Comment (sanderr) [reply src/extension.ts:164]: I see now it was there before your changes as well.

Comment (andraskvr) [reply src/extension.ts:164]: Yes, but now that you mention it, I'm not sure if it's necessary

File: src/extension.ts [hunk 2]
```diff
@@ -150,10 +145,35 @@ export function activate(context: ExtensionContext) {
 
 	const enable: boolean = workspace.getConfiguration('inmanta').ls.enabled
 
+	function createVenvIfNotExists() {
+		const pp: string = workspace.getConfiguration('inmanta').pythonPath;
+		if (pp && fs.existsSync(pp)) {
+			window.showInformationMessage(`Using Python Interpreter at ${pp}`);
```
Comment (wouterdb) [inline src/extension.ts:151]: This message will pop up every time the extensions starts?

File: src/extension.ts (hunk 1 above)
Comment (wouterdb) [reply src/extension.ts:164]: good point, should probably be removed, those who want --pre can do it by hand

Comment (wouterdb) [reply src/extension.ts:164]: (perhaps both pip-install's have to go into one method)

**Review by wouterdb — CHANGES_REQUESTED** [review]

**Review by sanderr — APPROVED** [review]

**Review by wouterdb — APPROVED** [review]

**Pull request merged**